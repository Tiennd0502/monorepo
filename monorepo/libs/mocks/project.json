{
  "name": "mocks",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/mocks/src",
  "projectType": "library",
  "tags": [],
  "// targets": "to see all targets run: nx show project mocks --web",
  "targets": {}
}
