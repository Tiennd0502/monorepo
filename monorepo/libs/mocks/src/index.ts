export * from './blog';
export * from './user';
export * from './orders';
export * from './product';
export * from './category';
