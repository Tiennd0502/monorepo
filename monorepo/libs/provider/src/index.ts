export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
