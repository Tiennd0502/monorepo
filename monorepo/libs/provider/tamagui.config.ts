import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
