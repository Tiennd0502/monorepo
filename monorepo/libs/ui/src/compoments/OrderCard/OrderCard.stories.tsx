import { Meta, StoryObj } from '@storybook/react';

import { ORDERS } from '@monorepo/constants';

import OrderCard from '.';

const meta: Meta<typeof OrderCard> = {
  title: 'Components/OrderCard',
  component: OrderCard,
};

export default meta;

type Story = StoryObj<typeof OrderCard>;

export const Default: Story = {
  args: {
    item: ORDERS[0],
    onViewDetail: (id: string) => null,
  },
};
