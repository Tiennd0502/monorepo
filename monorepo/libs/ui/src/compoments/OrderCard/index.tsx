import { memo, useCallback, useMemo } from 'react';
import { Stack, XStack } from 'tamagui';

// Types
import { ORDER_STATUS, Order } from '@monorepo/types';

// Constants
import { CURRENCY_UNIT } from '@monorepo/constants';

// Components
import Text from '../Text';
import Button from '../Button';
import Divider from '../Divider';

interface OrderCardProps {
  item: Order;
  onViewDetail: (value: string) => void;
}

const OrderCard = ({
  item: { id, quantity, totalAmount, createdAt, status },
  onViewDetail,
}: OrderCardProps) => {
  const handleViewDetail = useCallback(() => {
    onViewDetail(id);
  }, [id, onViewDetail]);

  const textColorForStatus = useMemo(() => {
    switch (status) {
      case ORDER_STATUS.Delivered.toString():
        return '$successPrimary';
      case ORDER_STATUS.Processing.toString():
        return '$warningPrimary';
      default:
        return '$errorPrimary';
    }
  }, [status]);

  return (
    <Stack borderRadius="$2.5" backgroundColor="$secondary">
      <XStack
        justifyContent="space-between"
        paddingTop="$3.75"
        paddingLeft="$5"
        paddingRight="$3.75"
        paddingBottom="$2.5"
      >
        <Text color="$primary" size="extraMedium">{`Order No${id}`}</Text>
        <Text color="$textLabel" size="medium">
          {createdAt}
        </Text>
      </XStack>
      <Divider height="$0.5" color="$backgroundTertiary" />
      <XStack
        justifyContent="space-between"
        paddingTop="$3.75"
        paddingLeft="$5"
        paddingRight="$3.75"
        paddingBottom="$2.5"
      >
        <XStack>
          <Text color="$textLabel" size="extraMedium">
            Quantity:&nbsp;
          </Text>
          <Text color="$primary" size="extraMedium">
            {quantity}
          </Text>
        </XStack>
        <XStack>
          <Text color="$textLabel">Total Amount:&nbsp;</Text>
          <Text
            color="$primary"
            fontWeight="bold"
          >{`${CURRENCY_UNIT}${totalAmount}`}</Text>
        </XStack>
      </XStack>
      <XStack
        justifyContent="space-between"
        paddingVertical="$5"
        paddingRight="$3.75"
      >
        <Button
          testID="view-detail"
          size="sm"
          width="$25"
          onPress={handleViewDetail}
        >
          Detail
        </Button>
        <Text color={textColorForStatus}>{ORDER_STATUS[+status]}</Text>
      </XStack>
    </Stack>
  );
};

export default memo(OrderCard);
