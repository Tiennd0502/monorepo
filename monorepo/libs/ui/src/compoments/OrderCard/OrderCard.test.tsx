import {fireEvent, render, waitFor} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import { ORDERS } from '@monorepo/mocks';
import OrderCard from '.';

const onViewDetail = jest.fn();

const baseProps = {
  item: ORDERS[0],
  onViewDetail,
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <OrderCard {...props} />
    </TamaguiProvider>
  );
};

describe('OrderCard', () => {
 
  it('should render OrderCard component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render OrderCard component with order status processing', () => {
    const container = setup({item: ORDERS[4]});

    expect(container).toMatchSnapshot();
  });

  it('should render OrderCard component with order status canceled', () => {
    const container =  setup({item: ORDERS[7]});

    expect(container).toMatchSnapshot();
  });

  it('calls onViewDetail when detail button clicked', () => {
    const {getByTestId} =  setup();

    waitFor(() => {
      const detailBtn = getByTestId('view-detail');
      fireEvent.press(detailBtn);

      expect(onViewDetail).toHaveBeenCalled();
    });
  });
});
