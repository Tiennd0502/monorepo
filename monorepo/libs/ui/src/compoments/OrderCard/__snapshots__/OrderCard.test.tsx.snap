// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCard should render OrderCard component 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "elevation": 16,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "overflow": "hidden",
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 16,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Order No01
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      20/03/2020
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f0f0f0",
        "height": 2,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#808080",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Quantity: 
      </Text>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        1
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#808080",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Total Amount: 
      </Text>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        $100
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingRight": 15,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      focusable={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#242424",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderStyle": "solid",
          "borderTopLeftRadius": 0,
          "borderTopRightRadius": 4,
          "borderTopWidth": 0,
          "color": "#fff",
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 7,
          "paddingTop": 7,
          "width": 100,
        }
      }
      testID="view-detail"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#fff",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Detail
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#27ae60",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      5
    </Text>
  </View>
</View>
`;

exports[`OrderCard should render OrderCard component with order status canceled 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "elevation": 16,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "overflow": "hidden",
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 16,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Order No08
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      21/03/2024
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f0f0f0",
        "height": 2,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#808080",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Quantity: 
      </Text>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        2
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#808080",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Total Amount: 
      </Text>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        $900
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingRight": 15,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      focusable={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#242424",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderStyle": "solid",
          "borderTopLeftRadius": 0,
          "borderTopRightRadius": 4,
          "borderTopWidth": 0,
          "color": "#fff",
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 7,
          "paddingTop": 7,
          "width": 100,
        }
      }
      testID="view-detail"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#fff",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Detail
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#eb5757",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      6
    </Text>
  </View>
</View>
`;

exports[`OrderCard should render OrderCard component with order status processing 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "elevation": 16,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "overflow": "hidden",
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 16,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Order No05
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      21/03/2024
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f0f0f0",
        "height": 2,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#808080",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Quantity: 
      </Text>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        1
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#808080",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Total Amount: 
      </Text>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        $600
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingRight": 15,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      focusable={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#242424",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderStyle": "solid",
          "borderTopLeftRadius": 0,
          "borderTopRightRadius": 4,
          "borderTopWidth": 0,
          "color": "#fff",
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 7,
          "paddingTop": 7,
          "width": 100,
        }
      }
      testID="view-detail"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#fff",
            "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Detail
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#f2c94c",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      3
    </Text>
  </View>
</View>
`;
